### ISSUE TITLE
reproducibility_tolerance: selection semantics, the ppm ceiling, and integer-label outputs are undocumented

### ISSUE THREAD

**Issue by Cyerme** [bioimage-io/spec-bioimage-io#770]
Follow-up to #659 ("add reproducibility_tolerance field in weight descriptions"),
which is still open and whose own TODO *"add validation for given output tensor
ids"* is still unchecked. This is not a request to reopen that discussion — the
field works. It is about what a packager cannot learn without reading the source.

## Problem

Packaging a model whose output is an **integer instance-label map**, we needed
`config.bioimageio.reproducibility_tolerance` to declare an accepted
cross-hardware deviation. Three things had to be recovered from the source
(`bioimageio/core/_resource_tests.py`), because they are not in the field docs:

**1. Selection is first-match-wins, not merge.** `_get_tolerance` iterates the
list and `break`s on the first entry matching both the weight format and the
output id:

```python
if (not a.weights_formats or wf in a.weights_formats) and (
    not a.output_ids or m in a.output_ids
):
    ...
    break
```

So a broad catch-all entry placed first silently shadows every later, more
specific entry. That is a reasonable design, but the ordering requirement is
load-bearing and undocumented.

**2. `mismatched_elements_per_million` is capped at 5000** (`Interval(ge=0,
le=5000)`, i.e. 0.5 %), and `relative_tolerance` at 0.01. Neither ceiling is
stated in the field description; you find out by getting a validation error.

**3. For integer outputs, `relative_tolerance` / `absolute_tolerance` are inert
and the ppm knob is the only usable one.** `evaluate_mismatched_elements` casts
both arrays to `float32` and applies `abs_diff > atol + rtol * abs(expected)`.
On a label map, an instance boundary that shifts by one pixel changes a label id
from e.g. 0 to 23 — arbitrarily far in both relative and absolute terms, no
matter how the tolerances are set. There is no rtol/atol setting between
"rejects a one-pixel boundary shift" and "accepts a wholly different
segmentation".

## Expected behavior

Documentation additions to the `reproducibility_tolerance` field (and to
#659 if it is still being revised):

- State that the first matching entry wins and that entries are therefore
  order-sensitive; recommend listing specific `output_ids` / `weights_formats`
  before any catch-all.
- State the bounds: `relative_tolerance` ≤ 0.01, `mismatched_elements_per_million`
  ≤ 5000, `absolute_tolerance` ≥ 0, and that defaults (1e-3 / 1e-3 / 100) are not
  serialised.
- Add a sentence for discrete-valued outputs: rtol/atol do not express anything
  useful on label maps, so use `mismatched_elements_per_million` alone and
  justify the number against a structural property of the output. (Example of
  what a reviewer can check: our allowance of 500 ppm is 508 px of 1 016 064,
  which is below the smallest instance in the test output at 2 907 px, so the
  allowance provably cannot conceal a gained, lost, split or merged instance.)

Optionally: validate that every `output_ids` entry names an output that exists
in the model — that is exactly the unchecked TODO in #659, and a typo there
currently makes the entry silently never match.

## Environment

- bioimageio.spec: 0.5.12.0
- bioimageio.core: 0.11.0
- Python: 3.11.16
- Model type: PyTorch (`pytorch_state_dict`), format_version 0.5.12, single
  integer-labelled output tensor

**Comment by FynnBe** [comment]
Thanks @Cyerme for this detailed issue! Appreciate your input!
I will:

- [x] update the documentation to point out the 'first match wins' behavior 
    - this is already the case, see https://bioimage-io.github.io/spec-bioimage-io/v0.5.14.1/api/bioimageio/spec/model/v0_5/?h=only+the+first+entry#bioimageio.spec.model.v0_5.BioimageioConfig.reproducibility_tolerance
- [x] and the value bounds.
- [x] ensure (default) tolerance values are always serialized so that possible future changes to the defaults do not cause third party implementations to deviate.
- [x] default to rtol/atol=0 for integer outputs (will give this another thought before)
- [x] fail validation for rtol/atol!=0 for integer outputs (will give this another thought before)
- [x] validate existence of `output_ids`

implemented in #771
